### DISCUSSION TITLE
Extending Micro-LC Library Components

### DISCUSSION THREAD

**Discussion by zachhutchins** [mia-platform/community#551] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

Version: 1.102.8

### Description

I know we are able to reference the base tool set for micro-lc, are the components offered in your BK library available to extend?

For example, I want to extend the BK-Modal component to make a custom modal.

What npm package do I need to bring in to have access to that?

**Comment by LiliumCandidum** [comment, marked as answer]
Hi @zachhutchins,

The back-kit library is not open source so it's not available for extensions or changes.

If you need a custom modal, you can create your own library with your web-components. It can be easily integrated in micro-lc.
Every back-kit component sends and listens to events through a common event bus and you can send/listen to the same events with your custom components to replace easily any of the back-kit components.

[Here](https://docs.mia-platform.eu/docs/microfrontend-composer/external-components/overview) you can find a guide to create your web-components library.
